programs/Makefile

```diff
@@ -55,12 +57,12 @@ APPS = \
 	pkey/rsa_sign_pss \
 	pkey/rsa_verify \
 	pkey/rsa_verify_pss \
-	psa/aead_demo \
-	psa/crypto_examples \
-	psa/hmac_demo \
-	psa/key_ladder_demo \
-	psa/psa_constant_names \
-	psa/psa_hash \
+	$(TF-PSA-CRYPTO_PROGRAM_PATH)/psa/aead_demo \
+	$(TF-PSA-CRYPTO_PROGRAM_PATH)/psa/crypto_examples \
+	$(TF-PSA-CRYPTO_PROGRAM_PATH)/psa/hmac_demo \
+	$(TF-PSA-CRYPTO_PROGRAM_PATH)/psa/key_ladder_demo \
+	$(TF-PSA-CRYPTO_PROGRAM_PATH)/psa/psa_constant_names \
+	$(TF-PSA-CRYPTO_PROGRAM_PATH)/psa/psa_hash \
```

Comment:
Could we avoid the change line 184 in `scripts/generate_visualc_files.pl` by using `../tf-psa-crypto/programs` here with a comment explaining why we do not use $(TF_PSA_CRYPTO_PROGRAMS_PATH) in that case?